packages/beacon-node/src/chain/bls/multithread/utils.ts

```diff
@@ -17,3 +19,16 @@ export function chunkifyMaximizeChunkSize<T>(arr: T[], minPerChunk: number): T[]
 
   return arrArr;
 }
+
+/**
+ * `rand` must not be exactly zero. Otherwise it would allow the verification of invalid signatures
+ * See https://github.com/ChainSafe/blst-ts/issues/45
+ */
+export function randomBytesNonZero(bytesCount: number): Buffer {
+  const rand = crypto.randomBytes(bytesCount);
+  for (let i = 0; i < bytesCount; i++) {
+    if (rand[0] !== 0) return rand;
```

Comment:
Why the loop? `i` is not accessed here. Or is it a time-attack protection?

Also, looks like this code comes from `blst` repo. Maybe it should be made available from there?

Reply:
```suggestion
    if (rand[i] !== 0) return rand;
```

Reply:
good catch!!! yes... this was copy/pasta from the blst library and i did not notice that bug before.  This should be updated in the base library as well.  I will put up a PR with that fix and export it so we can use it.  For now I have updated this to be functionally correct and once the PR from blst is exported we can remove this and use that one.